File: Python/Programmars/Level_1/자릿수 더하기.py
def solution(n):
    answer = 0
    for x in str(n):
        answer += int(x)
    return answer


print(solution(123))  # 6
print(solution(987))  # 24


# ref: str 변환 없이 각 자릿수 합 구하기
def sum_digit(number):
    if number < 10:
        return number
    return (number % 10) + sum_digit(number // 10)

File: JavaScript/Programmars/Level_1/자릿수 더하기.js
// 내 풀이 1: 몫과 나머지 연산
function solution(n) {
  let answer = 0;
  let tempN = n;
  while (tempN > 0) {
    answer += tempN % 10;
    tempN = Math.floor(tempN / 10);
  }

  return answer;
}

// 내 풀이 2: 문자열 변환
function solution(n) {
  let answer = 0;
  let strN = String(n);
  for (let num of strN) {
    answer += Number(num);
  }

  return answer;
}

console.log(solution(123)); // 6
console.log(solution(987)); // 24

// 다른 풀이: reduce 활용
function solution(n) {
  return String(n)
    .split('')
    .reduce((acc, val) => acc + parseInt(val), 0);
}
